scripts/installDependencies.sh

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/env bash
+
+
+TOOLS_BIN="${1?"Need to set full path to the location where binary tools will be downloaded"}"
+
+
+# the username used to access registry.redhat.io (OPTIONAL)
+RH_USER="${2:-}"
+# the password used to access registry.redhat.io (OPTIONAL)
+RH_PASSWORD="${3:-}"
+
+mkdir -p "${TOOLS_BIN}"
+
+# install upstream version of opm from github
+function installUptreamOpm()
+{
+    local version=${1:-"v1.15.1"}
+    local opmPrefix="opmup"
+    local platform=""
+    local architecture=""
+
+    # translate OSTYPE to expected platform name
+    if [[ "${OSTYPE}" == "linux-gnu"* ]]; then
+        platform="linux"
+    elif [[ "${OSTYPE}" == "darwin"* ]]; then
+        platform="darwin"
+    else
+        # unsupported os
+        return 1
+    fi
+
+    # translate HOSTTYPE to expected architecture name
+    if [[ "${HOSTTYPE}" == "x86_64" ]]; then
+        architecture="amd64"
+    elif [[ "${HOSTTYPE}" == "s390x" ]]; then
+        architecture="s390x"
+    elif [[ "${HOSTTYPE}" == "powerpc64le" ]]; then
+        architecture="ppc64le-opm"
+    else
+        # unsupported architecture
+        return 1
+    fi
+    
+    # download the upstream opm taking into account the version, platform, and architecture we're running on
+    assetURL=$(curl -s "https://api.github.com/repos/operator-framework/operator-registry/releases/tags/${version}" | jq --argjson platform "\"${platform}\"" --argjson architecture "\"${architecture}\"" -r '.assets[] | select(.browser_download_url | contains( ($platform + "-" + $architecture) ) ) | .url')
+    if ! "${TOOLS_BIN}/${opmPrefix}${version}" version 2>/dev/null && [[ -n ${assetURL} ]] ; then
+        curl -sL -H "Accept: application/octet-stream" -o "${TOOLS_BIN}/${opmPrefix}${version}" "${assetURL}" && chmod 755 "${TOOLS_BIN}/${opmPrefix}${version}"
+    fi
+
+}
+
+# installs downstream opm which requires docker for image extraction
+function installDownstreamOpm() 
+{
+    local opmCatalogBinaryImage="${1:-"registry.redhat.io/openshift4/ose-operator-registry:v4.6.0"}"
```

Comment:
we should take care not to pull downstream specific content into the upstream tests

Reply:
From previous comment:
> I removed the reliance on GitHub secrets but left in the code to allow for the possibility of downloading the downstream version of opm if we ever need to. That way a downstream fork of the tests could use the same script.

Was reluctant to pull out working code when it could be useful for a downstream fork to use when the time comes.